src/main/java/com/extremelyd1/game/progress/ProgressController.java

```diff
@@ -0,0 +1,45 @@
+package com.extremelyd1.game.progress;
+
+import org.bukkit.configuration.file.FileConfiguration;
+
+import java.util.List;
+
+public class ProgressController {
+
+    /**
+     * Rules for lines notifications
+     */
+    private List<String> linesRules;
+
+    /**
+     * Rules for full card count notifications
+     */
+    private List<String> fullCardCountRules;
+
+    /**
+     * Rules for lockout gamemode
+     */
+    private List<String> lockoutRules;
+
+    public ProgressController(FileConfiguration config) {
+        linesRules = config.getStringList("notify-progress.lines");
+        fullCardCountRules = config.getStringList("notify-progress.full-card");
+        lockoutRules = config.getStringList("notify-progress.lockout");
+    }
+
+    private Boolean notifyEachLine() {

```

Comment:
Preferably use the primitive `boolean` instead of the object wrapper, unless absolutely necessary. This holds for the other methods in this class as well.

Reply:
done